Cataloguer/MainWindow.xaml.cs

```diff
@@ -20,9 +22,38 @@ namespace Cataloguer
     /// </summary>
     public partial class MainWindow : Window
     {
+        public ICatalogueActions CatalogueActions { get; set; }
+
         public MainWindow()
         {
             InitializeComponent();
+            BusinessObjectLayer.ContainerManager.IntializeContainer();
+            ReadyCheck();
+        }
+
+        private void ReadyCheck()
+        {
+            if (CatalogueActions == null)
+            {
+                CatalogueActions = BusinessObjectLayer.ContainerManager.ProvideImplementation(CatalogueActions);
+            }
+
+            if (!CatalogueActions.IsApplicationReady())
```

Comment:
This is the main heart of what we were trying to do. Needed to clean up a bunch of other things though